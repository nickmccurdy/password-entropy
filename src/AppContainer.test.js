import { shallow } from 'enzyme'
import React from 'react'

import AppContainer from './AppContainer'

test('AppContainer', () => {
  const wrapper = shallow(<AppContainer />)
  expect(wrapper).toMatchSnapshot()
})

test('handleCustom', () => {
  const wrapper = shallow(<AppContainer />)
  wrapper.find('App').simulate('custom', { target: { value: '1' } })
  expect(wrapper.state()).toMatchObject({ options: { Custom: 1 } })
})

test('handleGeneric', () => {
  const wrapper = shallow(<AppContainer />)
  wrapper.find('App').simulate('generic', {
    target: { checked: false, name: 'Lowercase Letters' }
  })
  expect(wrapper.state()).toMatchObject({
    options: { Generic: { 'Lowercase Letters': false } }
  })
})

test('handleLength', () => {
  const wrapper = shallow(<AppContainer />)
  wrapper.find('App').simulate('length', { target: { value: '13' } })
  expect(wrapper.state('length')).toBe(13)
})

test('handleOptionsKey', () => {
  const wrapper = shallow(<AppContainer />)
  wrapper.find('App').simulate('optionsKey', null, 'Diceware')
  expect(wrapper.state('optionsKey')).toBe('Diceware')
})
