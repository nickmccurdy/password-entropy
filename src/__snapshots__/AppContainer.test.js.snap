// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer 1`] = `
<App
  length={12}
  onCustom={[Function]}
  onGeneric={[Function]}
  onLength={[Function]}
  onOptionsKey={[Function]}
  options={
    Object {
      "Custom": 0,
      "Generic": Object {
        "Lowercase Letters": true,
        "Numbers": true,
        "Symbols": true,
        "Uppercase Letters": true,
      },
    }
  }
  optionsKey="Generic"
/>
`;
